bitnami/cassandra/templates/statefulset.yaml

```
@@ -68,12 +68,29 @@ spec:
           image: {{ include "cassandra.volumePermissions.image" . }}
           imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
           command:
-            - /bin/bash
-            - -ec
+            - /bin/sh
+            - -cx
             - |
-              chown -R {{ .Values.containerSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }} /bitnami/cassandra
-          securityContext:
-            runAsUser: 0
+              {{- if .Values.persistence.enabled }}
+              {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
+              chown `id -u`:`id -G | cut -d " " -f2` /bitnami/cassandra
```

```suggestion
              chown `id -u`:`id -G | cut -d " " -f2` {{ .Values.persistence.path }}
```
How do you see replacing the hardcoded path by a customizable parameter? For that you need to create this parameter in the _values.yaml_/_values-production.yaml_, add it to the README table and replace `/bitnami/cassandra` by `{{ .Values.persistence.path }}` in the statefulset (there are 5 occurrences if I'm not wrong)

```console
$ ag '/bitnami/cassandra' templates
templates/statefulset.yaml
74:              chown -R {{ .Values.containerSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }} /bitnami/cassandra
82:              mountPath: /bitnami/cassandra
246:              mountPath: /bitnami/cassandra
249:              mountPath: /bitnami/cassandra/secrets
261:              mountPath: /bitnami/cassandra/conf
```

---

Absolutely! I'd probably go with `persistence.mountPath` to be consistent with the postgres chart, if that's ok?